File: results-05-12/typescript/typescript-00020-benchmark-store-website/qwen2-5-coder-32b/editor/final-src-code.ts

  // These environment variables are required, before any import.
  // Do not remove them, as they set environment variables for the Shinkai Tools.
  Deno.env.set('SHINKAI_NODE_LOCATION', "http://localhost:9950");
  Deno.env.set('BEARER', "debug");
  Deno.env.set('X_SHINKAI_TOOL_ID', "tool-id-debug");
  Deno.env.set('X_SHINKAI_APP_ID', "tool-app-debug");
  Deno.env.set('X_SHINKAI_LLM_PROVIDER', "o_qwen2_5_coder_32b");
  Deno.env.set('HOME', "results/typescript/typescript-00020-benchmark-store-website/qwen2-5-coder-32b/editor/home");
  Deno.env.set('MOUNT', "results/typescript/typescript-00020-benchmark-store-website/qwen2-5-coder-32b/editor/mount");
  Deno.env.set('ASSETS', "results/typescript/typescript-00020-benchmark-store-website/qwen2-5-coder-32b/editor/assets");
  
import { shinkaiSqliteQueryExecutor } from './shinkai-local-tools.ts';
import { getHomePath } from './shinkai-local-support.ts';

type CONFIG = {};
type INPUTS = { url: string };
type OUTPUT = { content: string, last_updated: string };

export async function run(config: CONFIG, inputs: INPUTS): Promise<OUTPUT> {
    const databaseName = 'default';
    const tableName = 'website_content';
    
    // Ensure the table exists
    await shinkaiSqliteQueryExecutor(databaseName, `
        CREATE TABLE IF NOT EXISTS ${tableName} (
            url TEXT PRIMARY KEY,
            content TEXT,
            last_updated DATETIME DEFAULT CURRENT_TIMESTAMP
        );
    `);

    // Fetch website content
    const response = await fetch(inputs.url);
    if (!response.ok) {
        throw new Error(`Failed to fetch the URL: ${inputs.url}`);
    }
    const content = await response.text();

    // Update or insert the content into the database
    await shinkaiSqliteQueryExecutor(databaseName, `
        INSERT INTO ${tableName} (url, content)
        VALUES (?, ?)
        ON CONFLICT(url) DO UPDATE SET content = ?, last_updated = CURRENT_TIMESTAMP;
    `, [inputs.url, content, content]);

    // Retrieve the updated row from the database
    const result = await shinkaiSqliteQueryExecutor(databaseName, `
        SELECT * FROM ${tableName} WHERE url = ?;
    `, [inputs.url]);

    return {
        content: result.result[0].content,
        last_updated: result.result[0].last_updated
    };
}

  
  // console.log('Running...')
  // console.log('Config: {}')
  // console.log('Inputs: {"url":"https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/test.html"}')
  
  try {
    const program_result = await run({}, {"url":"https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/test.html"});
    if (program_result) console.log(JSON.stringify(program_result, null, 2));
    else console.log(program_result);
  } catch (e) {
    console.log('::ERROR::', e);
  }

File: results-05-12/python/python-00009-benchmark-store-website/qwen2-5-coder-32b/editor/final-src-code.py

import os
import json
os.environ["SHINKAI_NODE_LOCATION"] = "http://localhost:9950"
os.environ["BEARER"] = "debug"
os.environ["X_SHINKAI_TOOL_ID"] = "tool-id-debug"
os.environ["X_SHINKAI_APP_ID"] = "tool-app-debug"
os.environ["X_SHINKAI_LLM_PROVIDER"] = "o_qwen2_5_coder_32b"
os.environ["HOME"] = "results/python/python-00009-benchmark-store-website/qwen2-5-coder-32b/editor/home"
os.environ["MOUNT"] = "results/python/python-00009-benchmark-store-website/qwen2-5-coder-32b/editor/mount"
os.environ["ASSETS"] = "results/python/python-00009-benchmark-store-website/qwen2-5-coder-32b/editor/assets"

from typing import Dict, Any
from shinkai_local_tools import shinkai_sqlite_query_executor
import requests

class CONFIG:
    pass

class INPUTS:
    url: str

class OUTPUT:
    table_data: List[Dict[str, Any]]

async def run(config: CONFIG, inputs: INPUTS) -> OUTPUT:
    # Fetch the content from the given URL
    response = requests.get(inputs.url)
    website_content = response.text

    # Define the database and table details
    database_name = 'default'
    table_name = 'website_content'

    # Create table if it doesn't exist
    create_table_query = f"""
    CREATE TABLE IF NOT EXISTS {table_name} (
        id INTEGER PRIMARY KEY AUTOINCREMENT,
        url TEXT NOT NULL UNIQUE,
        content TEXT NOT NULL
    );
    """
    shinkai_sqlite_query_executor(database_name, create_table_query, None)

    # Insert or update the website content in the table
    insert_update_query = f"""
    INSERT INTO {table_name} (url, content) 
    VALUES (?, ?)
    ON CONFLICT(url) DO UPDATE SET content = ?;
    """
    query_params = [inputs.url, website_content, website_content]
    shinkai_sqlite_query_executor(database_name, insert_update_query, query_params)

    # Select all data from the table
    select_query = f"SELECT * FROM {table_name};"
    result = shinkai_sqlite_query_executor(database_name, select_query, None)

    output = OUTPUT()
    output.table_data = result['result']
    return output

if __name__ == "__main__":
    import asyncio
    
    config = CONFIG()
    inputs = INPUTS()
    inputs.url="https://raw.githubusercontent.com/acedward/expert-octo-computing-machine/main/test.html"
    
    # Run the async function
    result = asyncio.run(run(config, inputs))
    print(json.dumps(result.__dict__))